File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

int GetCDCount(int num);
int solution(int left, int right) {
    int answer = 0;
    
    for(;left<=right;++left)
    {
        int CDCount = GetCDCount(left);
        if(CDCount % 2 == 0)
        {
            answer += left;
        }else
        {
            answer -= left;
        }
    }
    return answer;
}

int GetCDCount(int num)
{
    int half = sqrt(num)+1;
    int CDCount=0;
    for(int i=1;i<=half; ++i)
    {
        if(num % i == 0)
        {
            if(i != num/i)
                ++CDCount;
            ++CDCount;
        }
    }
    return CDCount;
}

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cs
using System;

public class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        
        for(int i = left; i<=right;++i)
        {
            int count = GetDivisorCount(i);
            if(count %2 == 0)
                answer += i;
            else
                answer -= i;
        }
        return answer;
    }
    
    public int GetDivisorCount(int target)
    {
        int sqrtTarget = (int)Math.Sqrt(target);
        int count=0;
        for(int i=1;i<=sqrtTarget;++i)
        {
            if(target % i == 0)
            {
                if(i != target/i)
                    ++count;
                ++count;
            }
        }
        
        return count;
    }
}